apps/src/blockly/addons/cdoFieldButton.js

```diff
@@ -45,25 +45,34 @@ export default class CdoFieldButton extends GoogleBlockly.Field {
   }
 
   /**
+   * Used to update the value of a field.
    * @override
+   * @param newValue The value to be saved.
    */
-  getValue() {
-    return String(this.value_);
+  doValueUpdate_(newValue) {
+    if (this.value_ !== newValue) {
+      if (this.changeHandler_) {
+        const override = this.changeHandler_(newValue);
+        if (override !== undefined) {
+          newValue = override;
+        }
+      }
+      this.value_ = newValue;
+    }
+    this.isDirty_ = true; // Block needs to be re-rendered.
```

Comment:
Do we always need to rerender, or only if the value has changed (ie, could this go inside of the top-most if statement, and set isDirty to false otherwise)?

Reply:
I agree that it makes sense to only need to re-render if value has changed. I don't think we need to set it to false because it is set to false after rendering [here](https://github.com/google/blockly/blob/139bc299a061c3d9fc81e8026d929cbe5589bb55/core/field.ts#L845). And it looks like `render` is [generally only called](https://github.com/google/blockly/blob/139bc299a061c3d9fc81e8026d929cbe5589bb55/core/field.ts#L835) in `getSize`.